src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+                || element.isAfterDeletionEnd() || element.isBeforeInsertion() || element.isAfterInsertion() || element.isNextToSoftClip();
+        final double referenceLikelihood;
+        final double nonRefLikelihood;
+        if (isAlt) {
+            nonRefLikelihood = QualityUtils.qualToProbLog10(qual);
+            referenceLikelihood = QualityUtils.qualToErrorProbLog10(qual) + MathUtils.LOG10_ONE_THIRD;
+            result.incrementNonRefAD(1);
+        } else {
+            referenceLikelihood = QualityUtils.qualToProbLog10(qual);
+            nonRefLikelihood = QualityUtils.qualToErrorProbLog10(qual) + MathUtils.LOG10_ONE_THIRD;
+            result.incrementRefAD(1);
+        }
+        // Homozygous likelihoods don't need the logSum trick.
+        result.increaseGenotypeLikelihood(0, referenceLikelihood + log10Ploidy);
+        result.increaseGenotypeLikelihood(likelihoodCount - 1, nonRefLikelihood + log10Ploidy);
+        // Heterozygous likelihoods need the logSum trick:
+        for (int i = 1, j = likelihoodCount - 2; i < likelihoodCount - 1; i++, j--) {
+            result.increaseGenotypeLikelihood(i,
+                    MathUtils.approximateLog10SumLog10(
+                            referenceLikelihood + MathUtils.log10(j),
+                            nonRefLikelihood + MathUtils.log10(i)));
+        }
+        if (isAlt && hqSoftClips != null && element.isNextToSoftClip()) {
+            hqSoftClips.increment(AlignmentUtils.calcNumHighQualitySoftClips(element.getRead(), HQ_BASE_QUALITY_SOFTCLIP_THRESHOLD));
+        }
+    }
+
+    /**
+     * Get a list of pileups that span the entire active region span, in order, one for each position
+     */
+    private List<ReadPileup> getPileupsOverReference(final Haplotype refHaplotype,
+                                                           final Collection<Haplotype> calledHaplotypes,
+                                                           final GenomeLoc paddedReferenceLoc,
+                                                           final AssemblyRegion activeRegion,
+                                                           final GenomeLoc activeRegionSpan,
+                                                           final ReadLikelihoods<Haplotype> readLikelihoods) {
+        if ( !calledHaplotypes.contains(refHaplotype)) throw new IllegalArgumentException("calledHaplotypes must contain the refHaplotype");
+        if ( readLikelihoods.numberOfSamples() != 1 ) throw new IllegalArgumentException("readLikelihoods must contain exactly one sample but it contained " + readLikelihoods.numberOfSamples());
+
+        final List<GATKRead> reads = activeRegion.getReads();
+
+        if ( debuggingWriter != null ) {
+            for (final GATKRead read : reads) {
+                debuggingWriter.addAlignment(read.convertToSAMRecord(activeRegion.getHeader()));
+            }
+        }
+
```

Comment:
I think this code below is a bit more complicate that what it needs to be.

So basically for position for which there are data as returned by ```libs``` we are putting the corresponding pileup in the output, for other we just make an empty one up. 

What about returning an fix size array (that can be encapsuated into a List with Arrays.asList if you want to keep the signature) and we have two loops. 

First one using libs to fill out the positions with data.
Second one set any remaining null position with the made-up empty pile-up.